### PR TITLE
Fix missing os-family="debian" in a bunch of packages

### PR DESCRIPTION
**Pull request by thomasblanc**
As discussed in #14906, this fixes the missing `os-family` problem that plagues raspbian-users.

The first commit was automatically generated by a script I made modifying the one by @toots [here](https://github.com/ocaml/opam-repository/pull/14440#issuecomment-508762509) ([my version](https://github.com/ocaml/opam-repository/files/3730078/odeb.txt) with a txt extension because github wanted me to).

The two big differences with toots's code: handling the case where several packages are required and modifying a lot of files inplace.

The second commit is a bunch of calls to `sed`:

```
s/os-distribution = "debian"/os-family = "debian" \& os-distribution != "ubuntu"/
s/ & os-distribution != "ubuntu" | os-distribution = "ubuntu"//

```

Some notes on where I'm not totally sure if this is a bug so I just leave it here just in case:
 * In `conf-efl.1.8`, the constraint on `debian` adds a constraint on `linux`. This might not work on `debian/kfreebsd` and maybe we should discourage that kind of pattern. (ping @axiles)
 * In `frama-c.10.0`, a conflict is noted on `debian`, there is no mention of `ubuntu` (and I excluded it to maintain behavior). (ping @bobot)
 * In several packages (take for example `bwrap.0.1`), there are constraints for some version(s) of `ubuntu` but no failure if on `ubuntu` but the version constraint fails. This is probably a bug.

Needless to say, testing that kind of constraints will be added to opam-lint shortly (ahem @AltGr).

@kit-ty-kate any thoughts?

### CODE REVIEW

File: packages/llvm/llvm.3.7/opam [hunk 1]
```diff
@@ -18,7 +18,7 @@ depends: [
   "ocamlbuild" {build}
 ]
 depexts: [
-  ["llvm-3.7-dev"] {os-distribution = "debian"}
+  ["llvm-3.7-dev"] {os-family = "debian" & os-distribution != "ubuntu"}
```
Comment (kit-ty-kate) [inline packages/llvm/llvm.3.7/opam]:
```suggestion
  ["llvm-3.7-dev"] {os-family = "debian"}
```
This should be fine without it

File: packages/llvm/llvm.3.6/opam [hunk 2]
```diff
@@ -17,7 +17,7 @@ depends: [
   "ocamlbuild" {build}
 ]
 depexts: [
-  ["llvm-3.6-dev"] {os-distribution = "debian"}
+  ["llvm-3.6-dev"] {os-family = "debian" & os-distribution != "ubuntu"}
```
Comment (kit-ty-kate) [inline packages/llvm/llvm.3.6/opam]:
```suggestion
  ["llvm-3.6-dev"] {os-family = "debian"}
```

File: packages/llvm/llvm.3.5/opam [hunk 3]
```diff
@@ -15,7 +15,7 @@ remove: [
   ["sh" "./compile.sh" "uninstall" "3.5" make prefix lib]
 ]
 depexts: [
-  ["llvm-3.5-dev"] {os-distribution = "debian"}
+  ["llvm-3.5-dev"] {os-family = "debian" & os-distribution != "ubuntu"}
```
Comment (kit-ty-kate) [inline packages/llvm/llvm.3.5/opam]:
```suggestion
  ["llvm-3.5-dev"] {os-family = "debian"}
```

File: packages/llvm/llvm.3.4/opam [hunk 4]
```diff
@@ -15,7 +15,7 @@ remove: [
   ["sh" "./compile.sh" "uninstall" "3.4" make prefix lib]
 ]
 depexts: [
-  ["llvm-3.4-dev"] {os-distribution = "debian"}
+  ["llvm-3.4-dev"] {os-family = "debian" & os-distribution != "ubuntu"}
```
Comment (kit-ty-kate) [inline packages/llvm/llvm.3.4/opam]:
```suggestion
  ["llvm-3.4-dev"] {os-family = "debian"}
```

File: packages/libsvm/libsvm.0.9.4/opam [hunk 5]
```diff
@@ -20,7 +20,7 @@ depends: [
   "lacaml"
   "jbuilder" {build & >= "1.0+beta2"}
 ]
```
Comment (kit-ty-kate) [inline packages/libsvm/libsvm.0.9.4/opam]:
```suggestion
  "conf-g++"
]
```
I think the depexts here can be replaced by this normal dependency

File: packages/libsvm/libsvm.0.9.4/opam [hunk 6]
```diff
@@ -20,7 +20,7 @@ depends: [
...
-depexts: ["g++"] {os-distribution = "debian"}
+depexts: ["g++"] {os-family = "debian" & os-distribution != "ubuntu"}
```
Comment (kit-ty-kate) [inline packages/libsvm/libsvm.0.9.4/opam]:
```suggestion
```

File: packages/frama-c/frama-c.10.0/opam [hunk 7]
```diff
@@ -66,7 +66,7 @@ patches: ["0004-Port-to-OCamlgraph-1.8.5.patch"]
 
 conflicts: [
   "why3" { >= "0.85" }
-  "ocaml" { = "4.03.0" & os-distribution = "debian" }
+  "ocaml" { = "4.03.0" & os-family = "debian" & os-distribution != "ubuntu" }
```
Comment (kit-ty-kate) [inline packages/frama-c/frama-c.10.0/opam]: This constraint seems really really weird but ok: it came from https://github.com/ocaml/opam-repository/pull/12861 and i think it's fine if it stayed with `os-distribution`

File: packages/conf-qt/conf-qt.5.2.1/opam [hunk 8]
```diff
@@ -28,20 +28,17 @@ post-messages:
     {os = "macos"}
 depexts: [
   ["qt5"] {os = "macos" & os-distribution = "homebrew"}
-  ["qt5-default" "qtdeclarative5-dev" "qt5-qmake"]
-    {os-distribution = "ubuntu"}
-  ["qt5-default" "qtdeclarative5-dev" "qt5-qmake"]
-    {os-distribution = "debian"}
-
-  # to set right paths for qt5 automatically we should
```
Comment (kit-ty-kate) [inline packages/conf-qt/conf-qt.5.2.1/opam]: Comments have been removed here. Could you put them back?

Comment (kit-ty-kate) [reply packages/conf-qt/conf-qt.5.2.1/opam]: Also changing the indentation as less as possible would be a bit better to review

File: packages/conf-protoc/conf-protoc.1.0.0/opam [hunk 9]
```diff
@@ -8,19 +8,17 @@ dev-repo: "git+https://github.com/protocolbuffers/protobuf.git"
...
-  ["libprotobuf-dev" "protobuf-compiler"]   {os-distribution = "ubuntu"}
-  ["protobuf-devel" "protobuf-compiler"]    {os-distribution = "centos"}
-  ["protobuf-devel" "protobuf-compiler"]    {os-distribution = "fedora"}
-  ["protobuf-devel" "protobuf-compiler"]    {os-distribution = "rhel"}
-  ["protobuf"]                              {os-distribution = "alpine"}
-  ["protobuf"]                              {os-distribution = "arch"}
-  ["protobuf-devel"]                        {os-family = "suse"}
-  ["protobuf"]                              {os = "freebsd"}
-  ["protobuf"]                              {os = "macos" & os-distribution = "homebrew"}
+  ["protobuf-devel" "protobuf-compiler"] {os-distribution = "centos"}
```
Comment (kit-ty-kate) [inline packages/conf-protoc/conf-protoc.1.0.0/opam]: Same here, I'd rather not change the indentation

File: packages/conf-protoc/conf-protoc.0.9/opam [hunk 10]
```diff
@@ -8,19 +8,17 @@ dev-repo: "git+https://github.com/protocolbuffers/protobuf.git"
...
-  ["protobuf-compiler"] {os-distribution = "ubuntu"}
   ["protobuf-compiler"] {os-distribution = "centos"}
   ["protobuf-compiler"] {os-distribution = "fedora"}
   ["protobuf-compiler"] {os-distribution = "rhel"}
-  ["protobuf"]          {os-distribution = "alpine"}
-  ["protobuf"]          {os-distribution = "arch"}
-  ["protobuf-devel"]    {os-family = "suse"}
-  ["protobuf"]          {os = "freebsd"}
-  ["protobuf"]          {os = "macos" & os-distribution = "homebrew"}
+  ["protobuf"] {os-distribution = "alpine"}
```
Comment (kit-ty-kate) [inline packages/conf-protoc/conf-protoc.0.9/opam]: same here

**Comment by kit-ty-kate** [comment]
Otherwise I think it's good

File: packages/frama-c/frama-c.10.0/opam (hunk 7 above)
Comment (thomasblanc) [reply packages/frama-c/frama-c.10.0/opam]: Seeing how specific this constraint is, it's probably safe either way :-). Reverting

**Comment by kit-ty-kate** [comment]
Your push-force erased previous changes. Could you put them back?

I encourage you to use `git push --force-with-lease` next time. It's a wonderful option, I still don't know why it's not the default but oh well.

**Comment by thomasblanc** [comment]
They're back!

I didn't know about `--force-with-lease`, this is an awesome option that I will try to use from now on.

**Comment by thomasblanc** [comment]
@kit-ty-kate I think we can merge now. Did I miss something?

**Comment by kit-ty-kate** [comment]
Oh I'm so sorry I missed your previous comment. Thanks!

**Pull request merged**